modules/apps/transfer/types/transfer_authz.go

```diff
@@ -114,3 +105,15 @@ func (a TransferAuthorization) ValidateBasic() error {
 	}
 	return nil
 }
+
+// isAllowedAddress returns a boolean indicating if the receiver address is valid for transfer.
+// gasCostPerIteration gas is consumed for each iteration.
+func isAllowedAddress(ctx sdk.Context, receiver string, allowedAddrs []string) bool {
+	for _, addr := range allowedAddrs {
+		ctx.GasMeter().ConsumeGas(gasCostPerIteration, "transfer authorization")
+		if addr == receiver {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
I moved this to the bottom of the file as it was in between multiple public methods.